### PR TITLE
refactor - switch active version definition

### PR DESCRIPTION
**Pull request by caiordjesus**
### What?
Make `switch_active_version` operation definition have only `"newVersion"`, instead of `"newScheduler"` and `"replacePods"`.

### Why?
Since now we have the "input" property in the Operation entity, and "input" is the representation of what the client gave to Maestro, we want the definition to be exactly this.
That said, the definition of "switch active version" should have only the version being switched to, and not the whole entity (since the user is not giving it to Maestro, is something that Maestro fetches from the database) and "replacePods" (which is a business rule of the operation itself)

### CODE REVIEW

File: internal/api/handlers/schedulers_handler_test.go [hunk 1]
```diff
@@ -1152,7 +1150,7 @@ func TestSwitchActiveVersion(t *testing.T) {
 		mux.ServeHTTP(rr, req)
 
 		require.Equal(t, 500, rr.Code)
-		require.Contains(t, rr.Body.String(), "failed to schedule operation: failed to schedule switch_active_version operation")
+		require.Contains(t, rr.Body.String(), "internal error")
```
Comment (guilhermocc) [inline internal/api/handlers/schedulers_handler_test.go:1153]: Shouldn't we return status code 404 here?

File: internal/core/operations/switch_active_version/switch_active_version_executor.go [hunk 2]
```diff
@@ -79,8 +79,18 @@ func (ex *SwitchActiveVersionExecutor) Execute(ctx context.Context, op *operatio
...
+
+	scheduler, err := ex.schedulerManager.GetSchedulerByVersion(ctx, op.SchedulerName, updateDefinition.NewActiveVersion)
+	if err != nil {
+		logger.Error("error fetching scheduler to be switched to", zap.Error(err))
+		return err
+	}
+
+	replacePods, err := ex.shouldReplacePods(ctx, scheduler)
+	if err != nil {
+		logger.Error("error calculating if should replace pods", zap.Error(err))
```
Comment (guilhermocc) [inline internal/core/operations/switch_active_version/switch_active_version_executor.go:91]:
```suggestion
		logger.Error("error deciding if should replace pods", zap.Error(err))
```

File: internal/core/operations/switch_active_version/switch_active_version_executor.go [hunk 3]
```diff
@@ -79,8 +79,18 @@ func (ex *SwitchActiveVersionExecutor) Execute(ctx context.Context, op *operatio
 	if !ok {
 		return fmt.Errorf("the definition is invalid. Should be type SwitchActiveVersionDefinition")
 	}
-	scheduler := &updateDefinition.NewActiveScheduler
-	replacePods := updateDefinition.ReplacePods
+
+	scheduler, err := ex.schedulerManager.GetSchedulerByVersion(ctx, op.SchedulerName, updateDefinition.NewActiveVersion)
```
Comment (guilhermocc) [inline internal/core/operations/switch_active_version/switch_active_version_executor.go:83]: We need to cover both of these new error scenarios with tests

File: internal/api/handlers/schedulers_handler_test.go (hunk 1 above)
Comment (guilhermocc) [reply internal/api/handlers/schedulers_handler_test.go:1153]: BTW, this change should not to be implemented in this PR

File: internal/core/services/scheduler_manager/scheduler_manager.go [hunk 4]
```diff
@@ -67,6 +72,18 @@ func (s *SchedulerManager) GetActiveScheduler(ctx context.Context, schedulerName
 	return activeScheduler, nil
 }
 
+func (s *SchedulerManager) GetSchedulerByVersion(ctx context.Context, schedulerName, schedulerVersion string) (*entities.Scheduler, error) {
```
Comment (guilhermocc) [inline internal/core/services/scheduler_manager/scheduler_manager.go:75]: Missing test for this new guy

**Review by guilhermocc — CHANGES_REQUESTED** [review]
Need to implement missing tests

File: internal/core/services/scheduler_manager/scheduler_manager.go [hunk 5]
```diff
@@ -49,6 +52,8 @@ type SchedulerManager struct {
 	logger           *zap.Logger
 }
 
+var _ ports.SchedulerManager = (*SchedulerManager)(nil)
```
Comment (arthur29) [inline internal/core/services/scheduler_manager/scheduler_manager.go:55]: Top irmão

File: internal/api/handlers/schedulers_handler_test.go (hunk 1 above)
Comment (caiordjesus) [reply internal/api/handlers/schedulers_handler_test.go:1153]: No, there's another test that returns 404 when the scheduler version is not found. This one simply reproduces an internal server error. See line 1126

**Review by guilhermocc — APPROVED** [review]

File: internal/core/operations/switch_active_version/switch_active_version_executor.go [hunk 6]
```diff
@@ -227,3 +237,11 @@ func (ex *SwitchActiveVersionExecutor) appendToNewCreatedRooms(schedulerName str
 func (ex *SwitchActiveVersionExecutor) clearNewCreatedRooms(schedulerName string) {
 	delete(ex.newCreatedRooms, schedulerName)
 }
+
+func (ex *SwitchActiveVersionExecutor) shouldReplacePods(ctx context.Context, newScheduler *entities.Scheduler) (bool, error) {
+	actualActiveScheduler, err := ex.schedulerManager.GetActiveScheduler(ctx, newScheduler.Name)
+	if err != nil {
+		return false, err
+	}
+	return actualActiveScheduler.IsMajorVersion(newScheduler), nil
```
[the remark this answers is not in the document]
Comment (caiordjesus) [reply internal/core/operations/switch_active_version/switch_active_version_executor.go:246]: It's a great idea! I just think it should be contained on another PR in the future... We'll come back to see this asap

**Pull request merged**